### ISSUE TITLE
Add Cuda 12.x support / Model for jGCaMP8s at 45Hz

### ISSUE THREAD

**Issue by sumiya-kuroda** [HelmchenLabSoftware/Cascade#74]
Hello again!

I am trying to train a model for our jGCaMP8s dataset (mouse cortex, imaged at 45Hz), and we are facing an issue that our cluster at our institute only supports CUDA 12.X. Are there anyone who managed to upgrade the Cascade pipeline to work with a newer version of TF? I was able to install Cascade on my local workstation via Docker, but unfortunately, it could not handle the training steps.

Adrian says I could ask here to train models for us, so is that also an option? Or is every user expected to train their own models by themselves, in general?

Best,
Sumiya

**Comment by sumiya-kuroda** [comment]
Relates to #52

**Comment by PTRRupprecht** [comment]
Hi @sumiya-kuroda,

I have not tested the Cascade pipeline with the newest versions of TF extensively. I believe that I started testing but then realized that there were backward and forward compatibility issues (existing pretrained models could not be used, and newly trained models could not be used by existing pipelines). If we see that it becomes crucial to have this compatibility, we can in the future set up an entirely new version of Cascade built around newer packages. But this would involve quite a bit of work, and I'm hesitant to invest it right now.

And yes, you can simply ask here for specific models, and they will be trained for you within a couple of days on our hardware (it might take a bit more time for next week specifically since I'm on holidays). I do this quite regularly, for example, yesterday for a model trained for 100 Hz imaging (issue #72). This is much easier than every user training their own models.

To request a new model, you only have to specify the parameters (frame rate, e.g., 45 Hz; the underlying ground truth, e.g., GCaMP8, or specifically GCaMP8s; and the "smoothing" parameter, which indicates the temporal precision you're aiming for). You can also just tell me about your application, and then I can help you decide about the parameters or train a couple of models which you can then try out and choose :-)

Best,
Peter

**Comment by sumiya-kuroda** [comment]
Hello Peter,

Yes, then could we ask you to train models for us, please? This is the parameter I was trying to use (but my workstation failed to run):

```py
cfg = dict( 
    model_name = 'GC8s_45Hz_smoothing100ms',    # Model name (and name of the save folder)
    sampling_rate = 45,    # Sampling rate in Hz (round to next integer)
    training_datasets = [
        'DS32-GCaMP8s-m-V1',
                        ],
    noise_levels = [noise for noise in range(2,30)],  # int values of noise values (do not use numpy here => representer error!)
    smoothing = 0.1,     # std of Gaussian smoothing in time (sec)
    causal_kernel = 0,   # causal ground truth smoothing kernel
    windowsize=70,
          )
```

I will use this model to infer spike rates from one of our datasets that we used RiboL1-jGCaMP8s in L2/3 of the mouse PL with a microprism implant. The noise level was relatively high, and that is why I put lots of noise values in `noise_levels`.

**Comment by PTRRupprecht** [comment]
Hi Sumiya @sumiya-kuroda 

Sounds good to me. I have started the training of a few models (one with `smoothing = 100 ms`, but also other models with less smoothening to take advantage of the high imaging rate) and will upload them and let you know once they are ready.

A comment: There is no ground truth for soma-localized RiboL1-jGCaMP8s - but I guess that the model trained on GCaMP8s data will still be the best match, given what I've heard from others about the kinetics of of RiboL1-GCaMP8. Unfortunately, I don't have experience with RiboL1-guided expression myself.

Best,
Peter

**Comment by PTRRupprecht** [comment]
Hi @sumiya-kuroda,

The models are now trained and available online:

- `GC8s_EXC_45Hz_smoothing100ms`
- `GC8s_EXC_45Hz_smoothing50ms`
- `GC8s_EXC_45Hz_smoothing20ms`

As you will see, I slightly adapted the training parameters (for example, bigger spacing of noise levels for noise levels >20). Let me know whether you encounter any specific problems with these models or if you would like to test models with slightly different training parameters!

Best,
Peter

**Comment by sumiya-kuroda** [comment]
Thank you very much, Peter. That was super quick! We are very excited!
